week-1-ts-ecosystem/w01e03-object-oriented-structure/src/types.ts

```diff
@@ -0,0 +1,24 @@
+export type ProductType = 'KUP TERAZ' | 'AUKCJA' | 'ODDAM ZA DARMO';
+
+export interface IPrice {
+  value: number;
+  currency: 'PLN' | 'USD' | 'EUR'; // Our shop handle only these currencies
+}
+
+export interface IProduct {
+  type: ProductType;
+  id: string;
+  name: string;
+  quantity: number;
+  price?: IPrice;
+}
+
+export interface ICart {
```

Comment:
Ten `interface` również jest niepotrzebny, powtarza elementy powiązane z `Cart`.

Zauważ że `class` to jednocześnie `typ` + implementacja
Więc samo `class` realizuje zadania `interface` - w tym kontekście przepisywanie tego do `I*` nie jest potrzebne.